src/ngraph/distributed/open_mpi.hpp

```diff
@@ -137,6 +137,41 @@ namespace ngraph
                 MPI_Bcast(in, count, data_type, root_id, MPI_COMM_WORLD);
             }
 
+            void recv(void* in, element::Type_t element_type, size_t count, int src_id) override
+            {
+                auto data_type = MPI_FLOAT;
+
```

Comment:
Should we also support integer type here?

Reply:
good point, add more data type support.